core/persist/viewmanager/ViewManagerModel.ts

```diff
@@ -278,18 +289,19 @@ export class ViewManagerModel<T extends PlainObject = PlainObject>
     }
 
     async selectViewAsync(token: string) {
-        // TODO - review if we benefit from async + masking - eg during intensive
-        //      component rebuild within setValue?
-        await wait();
-
-        this.selectedToken = token;
-
-        // Allow this model to restore its own persisted state in its ctor and note the desired
-        // selected token before views have been loaded. Once views are loaded, this method will
-        // be called again with the desired token and will proceed to set the value.
-        if (isEmpty(this.views)) return;
-
-        this.setValue(this.selectedView?.value ?? ({} as T));
+        // Introduce minimal wait and link to viewSelectionObserver to allow apps to mask.
+        await wait(100)
```

Comment:
@ghsolomon rightly pointed out that this random `100` value is unfortunate. I re-tested, and I found that I was unable to get the mask to reliably render in my app when swapping views unless I left this "high" value in place.  `await wait()` and `await wait(1)` did not do it.

(The app test case involve some heavyweight re-filtering of a large recordset + dashboard layout changes, but nothing that extreme - definitely a "real world" case.)

We discussed how ideally we could use some utility that perhaps keyed off of animation frames or otherwise was able to better detect that rendering had "settled down" - for want of a more technical term...